src/lib/isSlug.js

```diff
@@ -1,6 +1,6 @@
 import assertString from './util/assertString';
 
-let charsetRegex = /^[^\s-_](?!.*?[-_]{2,})[a-z0-9-\\][^\s]*[^-_\s]$/;
+let charsetRegex = /^(?!.*(?:-|_){2,})(?![-_])(?!.*(?:-|_)$)[a-z0-9*&\\]*(?:[-_][a-z0-9*&\\]+)*$/;
```

Comment:
Something else I noticed, but is there a reason this allows `&`, `*` and `\` special characters? The original allows `\` though I'm not sure why since Slugify ignores `\`, and @WikiRik mentioned that Slugify is the basis of the regex for isSlug. Slugify also ignores `*` and replaces `&` with 'and', so I would consider all of them to be invalid.